python/tank_vendor/shotgun_api3/shotgun.py

```diff
@@ -4090,7 +4091,18 @@ def _upload_data_to_storage(self, data, content_type, size, storage_url):
                     if e.code == 503:
                         raise ShotgunError("Got a 503 response when uploading to %s: %s" % (storage_url, e))
                     raise ShotgunError("Unanticipated error occurred uploading to %s: %s" % (storage_url, e))
-
+            except urllib.error.URLError as e:
+                if e.errno == errno.ETIMEDOUT:
+                    if attempt != max_attempts:
+                        time.sleep(float(attempt) * backoff)
+                        # If there is a timeout we will only allow one retry
+                        attempt = max_attempts
+                        continue
+                    else:
+                        raise ShotgunError("Error when uploading to %s: %s" % (storage_url, e))
+                else:
+                    LOG.error("Error when uploading to %s: %s" % (storage_url, e))
```

Comment:
Are you sure you want to log and not raise the exception? It was raising the exception before so this would be a new behvaior.

Reply:
I believe that before this exception was not raised and only executed line 4095

Reply:
No the else block is running only if no exception is raised. Any exception, not only the one captured by a except block